Sources/Web3Modal/Web3Modal.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+import UIKit
+
+#if SWIFT_PACKAGE
+public typealias VerifyContext = WalletConnectVerify.VerifyContext
+#endif
+
+/// Web3Modal instance wrapper
+///
+/// ```Swift
+/// let metadata = AppMetadata(
+///     name: "Swift dapp",
+///     description: "dapp",
+///     url: "dapp.wallet.connect",
+///     icons:  ["https://my_icon.com/1"]
+/// )
+/// Web3Modal.configure(metadata: metadata)
+/// Web3Modal.instance.getSessions()
+/// ```
+public class Web3Modal {
+    /// Web3Modalt client instance
+    public static var instance: Web3ModalClient = {
+        guard let config = Web3Modal.config else {
+            fatalError("Error - you must call Web3Modal.configure(_:) before accessing the shared instance.")
+        }
+        return Web3ModalClient(
+            signClient: Sign.instance,
+            pairingClient: Pair.instance as! (PairingClientProtocol & PairingInteracting)
+        )
+    }()
+    
+    struct Config {
+        let projectId: String
+        var sessionParams: SessionParams
+    }
+    
+    private(set) static var config: Config!
+
+    private init() {}
+
+    /// Wallet instance wallet config method.
+    /// - Parameters:
+    ///   - metadata: App metadata
+    public static func configure(
+        projectId: String,
+        metadata: AppMetadata,
+        sessionParams: SessionParams = .default
+    ) {
+        Pair.configure(metadata: metadata)
+        Web3Modal.config = Web3Modal.Config(projectId: projectId, sessionParams: sessionParams)
+    }
+    
+    public static func set(sessionParams: SessionParams) {
+        Web3Modal.config.sessionParams = sessionParams
+    }
```

Comment:
what the set method is for?
what's the difference between configure()?

Reply:
I see, does it work? It’s singletons and config is used on the instantiation, so with the first call only

Reply:
Yes, it works. Config is stored as a static variable, which can be changed only via these 2 methods. I'm calling `set()` from Dapp